syntax = "proto3";

option go_package = "github.com/lashajini/mind-palace/proto";

import "SharedTypes.proto";

service VDB {
  rpc Insert(InsertRequest) returns (SharedTypes.Empty) {}
  rpc Search(SearchRequest) returns (SearchResponse) {}
  rpc Ping(SharedTypes.Empty) returns (SharedTypes.Empty) {}
  rpc Drop(SharedTypes.Empty) returns (SharedTypes.Empty) {}
}

message InsertRequest {
  string user = 1;
  repeated VDBRow rows = 2;

  message VDBRow {
    string id = 1;
    string input = 2;
    string type = 3;
  }
}

message SearchRequest {
  string text = 1;
  string user = 2;
}

message SearchResponse {
  repeated VDBRow rows = 1;

  message VDBRow {
    string id = 1;
    float distance = 2;
    string type = 3;
  }
}
